sdk/agrifood/agrifood-farming-rest/review/agrifood-farming.api.md

```diff
@@ -208,40 +270,61 @@ export interface ApplicationDataListQueryParam {
 export interface ApplicationDataListQueryParamProperties {
     $maxPageSize?: number;
     $skipToken?: string;
-    associatedBoundaryIds?: string[];
-    ids?: string[];
+    associatedBoundaryIds?: Array<string>;
+    ids?: Array<string>;
     maxArea?: number;
     maxAvgMaterial?: number;
-    maxCreatedDateTime?: Date;
-    maxLastModifiedDateTime?: Date;
-    maxOperationEndDateTime?: Date;
-    maxOperationModifiedDateTime?: Date;
-    maxOperationStartDateTime?: Date;
+    maxCreatedDateTime?: Date | string;
+    maxLastModifiedDateTime?: Date | string;
+    maxOperationEndDateTime?: Date | string;
+    maxOperationModifiedDateTime?: Date | string;
+    maxOperationStartDateTime?: Date | string;
     maxTotalMaterial?: number;
     minArea?: number;
     minAvgMaterial?: number;
-    minCreatedDateTime?: Date;
-    minLastModifiedDateTime?: Date;
-    minOperationEndDateTime?: Date;
-    minOperationModifiedDateTime?: Date;
-    minOperationStartDateTime?: Date;
+    minCreatedDateTime?: Date | string;
```

Comment:
Feels pretty bad to have to constantly check to see if a value is a Date instance or a date-formatted string.

Reply:
Hi Will, we are generating this union for Input types only, Output is always a `string` because we cannot safely deserialize without runtime mapping*.

(*just a note that this is an RLC so we don't have mappers)

The reason these inputs are `string` or `Date` is to give the user a chance to work with them as it is more convenient for their use case. 

For example, if they are round-tripping data from and to the service it may be more convenient to just take the date as received from the service and put it in the request without having to convert it to a `Date`. On the other hand, if users are crafting a request from scratch, it may be more convenient to say `Date.now()` than crafting the date as a string.

I believe this has value, especially since it is for input types, users shouldn't need flow control to work with these types.

Reply:
Hi Will, we are generating this union for Input types only, Output is always a `string` because we cannot safely deserialize without runtime mapping*.

(*note that this is an RLC which don't have any runtime mapping)

The reason these inputs are `string` or `Date` is to give the user a chance to work with them as it is more convenient for their use case. 

For example, if they are round-tripping data from and to the service it may be more convenient to just take the date as received from the service and put it in the request without having to convert it to a `Date`. On the other hand, if users are crafting a request from scratch, it may be more convenient to say `Date.now()` than crafting the date as a string.

I believe this has value, especially since it is for input types, users shouldn't need flow control to work with these types.

Reply:
> it is for input types

I didn't catch this! If it's for inputs but not outputs then it doesn't feel bad anymore.

Reply:
I also didn't see that it was only for input types 😅

That's a cool feature though and I like it alot!